File: app/android/app/src/main/java/com/gonezo/multiplatform/audioextraction/application/AudioExtractionUseCase.java
package com.gonezo.multiplatform.audioextraction.application;

import com.gonezo.multiplatform.audioextraction.contract.ExtractionRequest;
import com.gonezo.multiplatform.audioextraction.contract.ExtractionResult;

public interface AudioExtractionUseCase {
  ExtractionResult execute(ExtractionRequest request);

  void cancel(String requestId);
}

File: core/bin/main/com/gonezo/audioextraction/application/usecase/AudioExtractionUseCase.kt
package com.gonezo.audioextraction.application.usecase

import com.gonezo.audioextraction.domain.contract.ExtractionRequest
import com.gonezo.audioextraction.domain.contract.ExtractionResult

interface AudioExtractionUseCase {
  fun execute(request: ExtractionRequest): ExtractionResult
  fun cancel(requestId: String)
}
